pkg/maps/nodemap/node_map_v2.go

```diff
@@ -149,80 +159,14 @@ func LoadNodeMapV2() (MapV2, error) {
 	return &nodeMapV2{bpfMap: bpfMap}, nil
 }
 
-// migrateV1 will migrate the v1 NodeMap to this NodeMapv2
-//
-// Ensure this always occurs BEFORE we begin handling K8s Node events or else
-// both this migration and the node events will be writing to the map.
-//
-// This migration will leave the v1 NodeMap preset on the filesystem.
-// This is due to some unfortunately versioning requirements which forced this
-// migration to occur within a patch release.
-//
-// When Cilium reaches v1.17 the v1 NodeMap will no longer be required and we
-// can unpin the map after migration.
-func (m *nodeMapV2) migrateV1(NodeMapName string, EncryptMapName string) error {
-	log.Debug("Detecting V1 to V2 migration")
-
-	// load v1 node map
-	nodeMapPath := bpf.MapPath(NodeMapName)
-	v1, err := ebpf.LoadPinnedMap(nodeMapPath)
-	if errors.Is(err, unix.ENOENT) {
-		log.Debug("No v1 node map found, skipping migration")
-		return nil
-	}
-	if err != nil {
-		return err
-	}
-	nodeMap := nodeMap{
-		bpfMap: v1,
-	}
-
-	// load encrypt map to get current SPI
-	encryptMapPath := bpf.MapPath(EncryptMapName)
-	en, err := ebpf.LoadPinnedMap(encryptMapPath)
-	if errors.Is(err, unix.ENOENT) {
-		log.Debug("No encrypt map found, skipping migration")
-		return nil
-	}
-	if err != nil {
-		return err
-	}
-	defer en.Close()
-
-	var SPI uint8
-	if err = en.Lookup(uint32(0), &SPI); err != nil {
-		return err
-	}
-
-	// reads v1 map entries and writes them to V2 with the latest SPI found
-	// from EncryptMap
-	count := 0
-	parse := func(k *NodeKey, v *NodeValue) {
-		v2 := NodeValueV2{
-			NodeID: v.NodeID,
-			SPI:    SPI,
-		}
-		count++
-		m.bpfMap.Put(k, &v2)
-	}
-
-	log.WithFields(logrus.Fields{
-		logfields.SPI: SPI,
-	}).Debugf("Migrated %v V1 node map entries to V2", count)
-
-	err = nodeMap.IterateWithCallback(parse)
-	if err != nil {
-		return fmt.Errorf("failed to iterate v1 node map %w", err)
-	}
-
-	return nil
-}
-
 func (m *nodeMapV2) init() error {
 	if err := m.bpfMap.OpenOrCreate(); err != nil {
 		return fmt.Errorf("failed to init bpf map: %w", err)
 	}
 
+	/* Clean up the v1 map. TODO remove this in v1.19. */
+	os.Remove(filepath.Join(bpf.TCGlobalsPath(), "cilium_node_map"))
+
```

Comment:
Is this the right place to clean up the old map?

Reply:
Maybe it would make sense to actually leave the `migrateV1` function, and have it remove NodeMapV1? It would signify that, this is the final stage of migrating to the v1 map and the entire function can be removed in v1.19?